lib/graphql/schema/argument.rb

```diff
@@ -121,6 +121,14 @@ def type
       # Used by the runtime.
       # @api private
       def prepare_value(obj, value)
+        # NOTE: This fixes the issue on queries.
+        # QUESTION: Should an InputObject #prepare method be called and then
+        # later the obj#prepare (GraphQL::Schema::Object) be called upon the
+        # InputObject prepared value? Or should it be one OR the another?
+        if value.respond_to?(:prepare)
+          value = value.public_send(:prepare)
```

Comment:
Since we're always calling the same method here, why not hardcode the method call? 

```suggestion
          value = value.prepare
```

Reply:
a2912dac627eeb61bd239b4e0a0e410900026829